traitlets/config/application.py

```diff
@@ -151,7 +151,7 @@ def _classes_inc_parents(self):
     argv = List()
 
     # Whether failing to load config files should prevent startup
-    raise_config_file_errors = Bool(False)
+    raise_config_file_errors = Bool(os.environ.get('TRAITLETS_APPLICATION_RAISE_CONFIG_FILE_ERROR', '').lower() == 'true')
```

Comment:
Any argument for making this a dynamic default so that the default value gets read at application construction time rather than at import time?  I imagine that would make this easier to test, if nothing else.